xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/commentfield.vm

```diff
@@ -0,0 +1,59 @@
+## ---------------------------------------------------------------------------
+## See the NOTICE file distributed with this work for additional
+## information regarding copyright ownership.
+##
+## This is free software; you can redistribute it and/or modify it
+## under the terms of the GNU Lesser General Public License as
+## published by the Free Software Foundation; either version 2.1 of
+## the License, or (at your option) any later version.
+##
+## This software is distributed in the hope that it will be useful,
+## but WITHOUT ANY WARRANTY; without even the implied warranty of
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+## Lesser General Public License for more details.
+##
+## You should have received a copy of the GNU Lesser General Public
+## License along with this software; if not, write to the Free
+## Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+## 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+## ---------------------------------------------------------------------------
+###
+### Return a single comment field
+###
+###
+## requested field name 
+#set ($name = $request.name)
+## the number of the comment, if no number is provided we generate a field for a new comment.
+#set ($number = $numbertool.toNumber($request.number).intValue())
+
+## the 'comment' field from the XWikiComments class is used as a template for the field to display
+#set ($xCommentClass = 'XWiki.XWikiComments')
+#set ($commentClass = $xwiki.getDocument($xCommentClass).getxWikiClass())
+
+## the number of the comment, if no number is provided we generate a field for a new comment.
+#if ($number && $number > -1)
+  #set ($commentObj = $doc.getObject($xCommentClass, $number))
+  #set ($value = $commentObj.getValue('comment'))
+#else
+  #set ($commentObj = $doc.newObject($xCommentClass))
+  #set ($value = '')
+#end
+
+## We need to know the type of editor used for comments.
+#set ($defaultEditorId = $services.edit.syntaxContent.defaultEditorId)
+<input type='hidden' name='defaultEditorId' value="$defaultEditorId" />
+
+## copy and rename to the requested name of the comment field
```

Comment:
OK, I will try my best to improve explanations there, but I didn't follow much so maybe I'll let @manuelleduc improves it when he'll come back :)

Reply:
So actually I'm not comfortable on commenting those lines: I don't really understand why those manipulations are performed on the XClass... I understand that the goal of the file is to display a specific XWikiComment xobject field in edit.
Now I really don't get those lines: 
```
#set ($ocls = $commentObj.getxWikiClass())
#set ($discard = $ocls.getXWikiClass().put($name, $commentClass.getXWikiClass().get('comment')))
#set ($discard = $ocls.getXWikiClass().get($name).setName($name))
```

Reply:
The name comes from ``comments.js``' ``reloadEditor`` function:

```
var name = 'XWiki.XWikiComments_comment';
if (commentNbr) {
  name = 'XWiki.XWikiComments_' + commentNbr + '_comment';
}
```
As I mentioned above, we're supporting 2 use cases:

* editing a comment, in which case the field name looks like ``XWiki.XWikiComments_0_comment``
* adding a new comment, in which case the field name looks like ``XWiki.XWikiComments_comment``; we can't use an object number because multiple users can add comments at the same time

The first use case is easy. You just need to call ``$doc.display()``. The second case is a bit tricky because we want to display a comment without having a comment. So the first thing we do is we add a new temporary (unsaved) comment object, see above:

    #set ($commentObj = $doc.newObject($xCommentClass))

But we can't simply call ``$doc.display()`` on this object because we would get ``XWiki.XWikiComments_N_comment`` as the field name, where N is the number of the new temporary object. Instead we want the field name to be ``XWiki.XWikiComments_comment``. To achieve this we add a new property to the comment class by copying the comment property and changing the name. Then we display this new property.

Reply:
Thanks for the explanation, so if I follow you, all that stuff we're doing before the display we normally need to do it only in case of newObject, so in the same `#else` than in https://github.com/xwiki/xwiki-platform/pull/1317/files/4bd7613af141305df55f32d398206afed5d8c329#diff-118419b8c5fd09db95b383d36464a362R38 right? 

That would feel cleaner to me to only do it in that case wdyt?

Reply:
Yes, we should do it only when there's no comment number specified.

Reply:
OK going to clarify the code then.

Reply:
> Yes, we should do it only when there's no comment number specified.

So actually we were wrong about that: Manuel was changing the name for both editing and adding, because else we'd obtain the field name `comment` and not `XWiki.XWikiComments_N_comment`. So the problem is that in case of editing you won't send the complete name of the comment to the server. 
Going to revert my change, I'm just improving comments and variable names there.